addons/sale/models/sale_order_line.py

```diff
@@ -616,12 +616,31 @@ def _prepare_procurement_values(self, group_id=False):
         """
         return {}
 
-    @api.depends('product_id')
+    def _is_downpayment_draft(self):
+        # Verify that this SO line is
+        # first a down payment and
+        # second that the invoice(s) line are in a draft state
+        self.ensure_one()
+
+        if self.is_downpayment:
+            for line in self._get_invoice_lines():
+                if line.move_id.state != 'draft':
+                    return False
+                return True
+        return False
```

Comment:
Prefer to check conditions and return the result directly

```suggestion
        if not self.is_downpayment:
            return False
           
        for line in self._get_invoice_lines():
            if line.move_id.state != 'draft':
                return False
        
        return True
```

Reply:
done :)